MotorOutput implementation for elevator

haydenheroux opened the pull request.

lddtwo456 committed:
MotorOutput Implementation

lddtwo456 commented:
Replaced PositionController code in the Elevator subsystem with MotorOutput, and also introduced the more useful fading in of PID voltage from pr #18

haydenheroux commented on src/main/java/frc/robot/elevator/Elevator.java:
```diff
@@ -174,19 +215,39 @@ public void fastPeriodic() {
+      voltageOut.mut_replace(feedforwardVolts + feedbackVolts, Volts);
+    } else {
+      // Otherwise, dont change voltageOut at all, and reset voltageSet
+      voltageSet = false;
```
This will set voltageSet to false if voltageSet was true. Is this the intended effect?

lddtwo456 committed:
calculation clarification (that rhymes)

lddtwo456 commented:
I split the calculation for fading in PID voltage into a few steps for clarification purposes, there are still some other things I want to fix like making the state of using a manually set voltage vs calculating a voltage with feedforward and feedback stuff a bit less awful, but this should at least be enough to test stuff out at the meeting today. Will be working on cleaning stuff up though, and won't merge until it's actually decently written.

haydenheroux commented:
Looks good to me, feel free to merge either before or after testing. Good luck at the meeting!

lddtwo456 committed:
meeting testing

not able to tune everything, but pure ff control works well with this structure

lddtwo456 committed:
voltageSet clarification?

should clear up why voltageSet (now manualVoltageSet) is being used how it is being used

lddtwo456 committed:
fix logging weirdness

lddtwo456 committed:
meeting tuning (works great wow)

lddtwo456 committed:
Improved logging

lddtwo456 committed:
Merge branch 'MotorOutput-implementation-for-elevator' of https://github.com/Gongoliers/Robot2025 into MotorOutput-implementation-for-elevator
